### ISSUE TITLE
`modules.builtin` empty after adding NVIDIA drivers to installer image

### ISSUE THREAD

**Issue by rtrox** [siderolabs/talos#6681]
### Description

Currently, the way the installer target in the dockerfile is set up, it copies files in from any `customization`s, overwriting any files present with the same name. This is problematic for customizations like the NVIDIA Proprietary drivers, as the Pkgfile which generates those drivers generates a blank `modules.builtin` file in /lib/modules. As such, when you follow the [NVIDIA Proprietary Drivers Instructions](https://www.talos.dev/v1.2/talos-guides/configuration/nvidia-gpu-proprietary/) to the letter, the generated image will have the NVIDIA drivers, but all other built-ins (such as iptables) will be unavailable.

### Logs

No relevant logs to speak of, but I confirmed this in a few ways:
1. Simplest: `talosctl read /lib/modules/5.15.83-talos/modules.builtin` returns no output on a node upgraded with an nvidia image.
2. Less Simple: I downloaded the default installer, and the compiled nvidia installer, and compared by expanding and unsquashing the initramfs in each image. In the nvidia built installer, `modules.builtin` was empty.
3. Confirmation: See the Workaround below, which fixed the inability to utilize iptables by editing the customization image prior to build.

Can definitely give a more complete repro if needed.

### Workaround
I worked around this by removing the `modules.builtin.*` files from my customization prior to build, like this:
```
FROM alpine:latest as customization_prep
COPY --from=ghcr.io/rtrox/nonfree-kmod-nvidia:v1.3.0-5-g6509d23 /lib/modules /tmp/lib/modules
RUN find /tmp/ -name modules.builtin* -exec rm {} \;

FROM scratch as customization
COPY --from=customization_prep /tmp/lib/modules /lib/modules

FROM ghcr.io/siderolabs/installer:v1.3.0
COPY --from=ghcr.io/rtrox/kernel:v1.3.0-5-g6509d23 /boot/vmlinuz /usr/install/${TARGETARCH}/vmlinuz
```
If working around this from scratch, it's likely easier to just remove this line from the `pkgs` repo prior to building:
https://github.com/siderolabs/pkgs/blob/6509d23595cf1862cd80728500b230fcff67ff98/nonfree/kmod-nvidia/pkg.yaml#L40

### Environment

- Talos version:
```
Server:
        NODE:        REDACT
        Tag:         v1.3.0
        SHA:         323e3ccd
        Built:
        Go version:  go1.19.4
        OS/Arch:     linux/amd64
        Enabled:     RBAC
```
- Kubernetes version: 1.26.0
- Platform: amd64

**Comment by rtrox** [comment]
I'd also put this together for the folks in k8s-at-home as documentation of the build & workaround. It might be good to pull in the versioning pointers to the upstream docs (I can likely do this tomorrow).
https://gist.github.com/rtrox/fb6dfca524b396566b65d7653eb45037

**Comment by frezbo** [comment]
Fixed by #6751
